### PR TITLE
test all configured detector logics, not just the first

### PR DESCRIPTION
**Pull request by emiltin**
a PR makes it easier to follow the work in the branch

### CODE REVIEW

File: spec/support/command_helpers.rb [hunk 1]
```diff
@@ -502,22 +502,22 @@ def switch_input indx
...
+        [{'sCI'=>'S0002','n'=>'detectorlogicstatus','s'=>/^.{#{indx}}1/}]
+      )
+      
+      force_detector_logic component, mode:'False'
+      Validator.config['main_component'] = Validator.config['main_component']
+      wait_for_status(@task,
+        "deactivate detector logic #{component}",
+        [{'sCI'=>'S0002','n'=>'detectorlogicstatus','s'=>/^.{#{indx}}0/}]
+      )
+      indx = indx + 1
```
Comment (emiltin) [inline spec/support/command_helpers.rb:519]: you can use each_with_index, then you don't have to manually count

File: spec/support/command_helpers.rb [hunk 2]
```diff
@@ -502,22 +502,22 @@ def switch_input indx
...
-    )
-    
-
-    force_detector_logic component, mode:'False'
-    Validator.config['main_component'] = Validator.config['main_component']
-    wait_for_status(@task,
-      "deactivate detector logic #{component}",
-      [{'sCI'=>'S0002','n'=>'detectorlogicstatus','s'=>/^.{#{indx}}0/}]
-    )
+    Validator.config['components']['detector_logic'].keys.each do |component|
```
Comment (emiltin) [inline spec/support/command_helpers.rb:505]: use each_with_index

**Comment by emiltin** [comment]
looks good to me.
are there other helper methods that uses detector logics, that should be updated

File: spec/support/command_helpers.rb (hunk 1 above)
Comment (otterdahl) [reply spec/support/command_helpers.rb]: Should be fixed now

**Comment by mikaelbroms** [comment]
'forcing is set with M0019' works with 2 pcs detector logics KK + AG9998 = 001DL001: KK + AG9998 = 001DL002:

if only force_detector_logic component, mode: 'True' runs in
def switch_detector_logic

if also force_detector_logic component, mode: 'False' runs in
def switch_detector_logic

then it does not work with one or more detector logics

**Comment by mikaelbroms** [comment]
It seems that 'force_detector_logic component, mode:'False' not work in TLC.
Or that the subscription
wait_for_status
 [{'sCI'=>'S0002','n'=>'detectorlogicstatus','s'=>/^.{#{indx}}0/}]
not goes to zero within the timeout

**Comment by otterdahl** [comment]
I can confirm that is doesn't work. Might be because of some configuration setting in the TLC

**Comment by emiltin** [comment]
Is there more to do here? It seems you have confirmed that the problem is in th TLC. In that case I think we can delete this PR without merging?

**Comment by mikaelbroms** [comment]
The purpose for this PR is to test multi detector logics. The code for testning multi detektor logics is working. 

Skickat från min iPhone

**Comment by otterdahl** [comment]
I think this PR works. I'm merging this

**Pull request merged**